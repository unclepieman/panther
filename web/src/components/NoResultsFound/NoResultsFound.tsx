/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { Text, Flex, Img } from 'pouncejs';
import NothingFound from 'Assets/illustrations/nothing-found.svg';

interface NoResultsFoundProps {
  title?: string;
}

const NoResultsFound: React.FC<NoResultsFoundProps> = ({ title = 'No Results' }) => {
  return (
    <Flex justify="center">
      <Flex
        direction="column"
        align="center"
        justify="center"
        backgroundColor="navyblue-500"
        borderRadius="circle"
        width={260}
        height={260}
      >
        <Img
          ml={6}
          nativeWidth={95}
          nativeHeight={90}
          alt="Document and magnifying glass"
          src={NothingFound}
        />
        <Text color="navyblue-100" fontWeight="bold" mt={2}>
          {title}
        </Text>
      </Flex>
    </Flex>
  );
};

export default NoResultsFound;
