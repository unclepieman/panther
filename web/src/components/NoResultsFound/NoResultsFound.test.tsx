/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { render } from 'test-utils';
import NoResultsFound from './NoResultsFound';

describe('NoResultsFound', () => {
  it('matches snapshot', () => {
    const { container } = render(<NoResultsFound />);
    expect(container).toMatchSnapshot();
  });

  it('contains proper semantics', () => {
    const { getByText, getByAltText } = render(<NoResultsFound />);
    expect(getByAltText('Document and magnifying glass')).toBeInTheDocument();
    expect(getByText('No Results')).toBeInTheDocument();
  });

  it('allows a user to override the text', () => {
    const { getByText } = render(<NoResultsFound title="Fake title" />);
    expect(getByText('Fake title')).toBeInTheDocument();
  });
});
