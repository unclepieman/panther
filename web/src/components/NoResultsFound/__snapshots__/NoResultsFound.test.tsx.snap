// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResultsFound matches snapshot 1`] = `
.panther-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.panther-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #1d2a3f;
  border-radius: 99999px;
  width: 260px;
  height: 260px;
}

.panther-0 {
  max-width: 100%;
  margin-left: 24px;
}

.panther-1 {
  color: #86a3c3;
  font-weight: 700;
  margin-top: 8px;
}

<div>
  <header
    id="main-header"
  />
  <div
    class="panther-3"
  >
    <div
      class="panther-2"
    >
      <img
        alt="Document and magnifying glass"
        class="panther-0"
        height="90"
        src="test-file-stub"
        width="95"
      />
      <p
        class="panther-1"
      >
        No Results
      </p>
    </div>
  </div>
  <footer
    id="footer"
  />
</div>
`;
